pkg/preference/preference_test.go

```diff
@@ -491,6 +491,33 @@ func TestSetConfiguration(t *testing.T) {
 			existingConfig: Preference{},
 			wantErr:        true,
 		},
+		{
+			name:           fmt.Sprintf("Case 27: set %s to non bool value", ConsentTelemetrySetting),
+			parameter:      ConsentTelemetrySetting,
+			value:          "123",
+			existingConfig: Preference{},
+			wantErr:        true,
+		},
+		{
+			name:           fmt.Sprintf("Case 28: set %s from nil to true", ConsentTelemetrySetting),
+			parameter:      ConsentTelemetrySetting,
+			value:          "true",
+			existingConfig: Preference{},
+			wantErr:        false,
+			want:           "true",
+		},
+		{
+			name:      fmt.Sprintf("Case 29: set %s from true to false", ConsentTelemetrySetting),
+			parameter: ConsentTelemetrySetting,
+			value:     "false",
```

Comment:
We don't want an error in this test case. But the value is a string. Are we sure we don't want an error in this one?

Reply:
I think it will be fine, the string will be parsed to bool in  `SetConfiguration`. But I guess I can make it bool instead.

Reply:
https://github.com/openshift/odo/pull/4482/commits/b4d64b2865c1806d7431595c5bed22935c258f6f